+ Recent changes relevant to indexes (#5858)
edgedb/edgedb: docs/datamodel/indexes.rst

Indexes
=======

> An index is a data structure used internally to speed up filtering, sorting,

Reviewer: ```suggestion
An index is a data structure used internally to speed up filtering, ordering,
```

and grouping operations. Most commonly, indexes are declared within object
type declarations and reference a particular property; this will speed up
any query that references that property in a ``filter``, ``order by``, or
``group`` clause.

.. note::

  While improving query performance, indexes also increase disk and memory
  usage and slow down insertions and updates. Creating too many indexes may be

>   detrimental; only index properties you often filter, ordero or group by.

Reviewer 2: ```suggestion
  detrimental; only index properties you often filter, order, or group by.
```

      index on (.name);
    }

> By indexing on ``User.name``, queries that filter, order, or group by the 

Reviewer 2: Oh, I just noticed that we say "order" here, but "sort" in the other two places. I kinda like "order" since that's the related keyword, even though "sort" is probably the better verb. At any rate, just being consistent is all I really wanted to call out here, so whichever direction makes the most sense to y'all is good with me.

``name`` property will be faster, as the database can look up a name in 

> the index instead of scanning through all Users sequentially.

Reviewer: ```suggestion
the index instead of scanning through all ``User`` objects sequentially.
```

----------------------------

A *composite index* is an index that references multiple properties. This will

> speed up queries that filter, sort, or group on *both properties*. In EdgeDB,

Reviewer: ```suggestion
speed up queries that filter, order, or group on *both properties*. In EdgeDB,
```
